smarts/core/mission_planner.py

```diff
@@ -150,62 +177,78 @@ def closest_point_on_lane(self, position, lane_id: str):
         lane = self._road_network.lane_by_id(lane_id)
         return self._road_network.lane_center_at_point(lane, position)
 
-    def waypoint_paths_at(self, sim, pose: Pose, lookahead: float):
-        """Call assumes you're on the correct route already. We do not presently
-        "replan" in case the route has changed.
-        """
-        assert (
-            self._did_plan
-        ), "Must call plan(...) before being able to invoke the mission planner."
-
-        edge_ids = self._edge_ids(pose)
-        if edge_ids:
-            return self._waypoints.waypoint_paths_along_route(
-                pose.position, lookahead, edge_ids
-            )
+    def waypoint_paths_at(
+        self,
+        pose: Pose,
+        lookahead: int,
+        within_radius: float = 5,
+        filter_from_count: int = 3,
+        constrain_to_route: bool = True,
+    ) -> List[List[Waypoint]]:
```

Comment:
I agree.  In this PR, I was trying not to change any existing behavior / contracts.  We have been using ~1m spacing for waypoints/lanepoints thus far, so I kept that (and didn't change an lookahead parameters either).   Since there currently appear to be two main uses for `Waypoints` -- sent to agents via sensors and used in the lane-following controller -- If/when we have this discussion, it will need to be with existing users (e.g., Ultra) to ensure that we don't break things for their agents.